common/base/src/net.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::net::Ipv4Addr;
+use std::net::SocketAddrV4;
+use std::net::TcpListener;
+use std::net::UdpSocket;
+
+pub fn get_free_tcp_port() -> u16 {
+    // Assigning port 0 requests the OS to assign a free port
+    let loopback = Ipv4Addr::new(127, 0, 0, 1);
```

Comment:
Can we strip out `1-1024` to avoid permission denied error on the Linux platform?

Reply:
> Bind the socket to port 0. A random free port from 1024 to 65535 will be selected.

It looks ok

Reply:
It's automatically by the config file `/proc/sys/net/ipv4/ip_local_port_range `,  see [QA](https://unix.stackexchange.com/questions/55913/whats-the-easiest-way-to-find-an-unused-local-port)

Reply:
Oh, I learned something new.